File: Python/Semi-Colon E-commerce/Store.py
from Customer import Customer

class Store:
    def __init__(self, store_name, store_branch_address, store_telephone, products, cashiers):
        self._store_name = store_name
        self._store_branch_address = store_branch_address
        self._store_telephone = store_telephone
        self._products = products
        self._cashiers = cashiers

    def get_store_name(self):
        return self._store_name

    def get_store_branch(self):
        return self._store_branch_address

    def get_store_telephone(self):
        return self._store_telephone

    def login(self, cashier_id, cashier_password):
        for cashier in self._cashiers:
            if cashier.get_cashier_id() == cashier_id and cashier.get_cashier_password() == cashier_password:
                return cashier
        return None

    def checkout(self, customer_name):
        return Customer(customer_name)

File: Java/Semi-Colon E-commerce/src/Store.java
import java.util.ArrayList;

public class Store {
    private final String storeName;
    private final String storeBranchAddress;
    private final String storeTelephone;
    private final ArrayList<Product> products;
    private final ArrayList<Cashier> cashiers;

    public Store(String storeName,  String storeBranchAddress, String storeTelephone, ArrayList<Product> products, ArrayList<Cashier> cashiers) {
        this.storeName = storeName;
        this.storeBranchAddress = storeBranchAddress;
        this.storeTelephone = storeTelephone;
        this.products = products;
        this.cashiers = cashiers;
    }

    public String getStoreName() {
        return storeName;
    }

    public String getStoreBranch() {
        return storeBranchAddress;
    }


    public String getStoreTelephone() {
        return storeTelephone;
    }

    public Cashier login(int cashierID, String cashierPassword) {
        for (Cashier cashier : cashiers) {
            if (cashier.getCashierID() == cashierID && cashier.getCashierPassword().equals(cashierPassword)) {
                return cashier;
            }
        }
        return null;
    }

    public Customer checkout(String customerName) {
        return new Customer(customerName);
    }

}
